### DISCUSSION TITLE
Strange remove behavior

### DISCUSSION THREAD

**Discussion by erikhermansson79** [wixtoolset/issues#7180] [Questions]
If I run my MSI-file, when the product is installed, and choose "Remove" then nothing is removed, but if I instead run "msiexec /x" everything is removed correctly. If I choose "Change" and disable a feature it is not removed either. Maybe those issues are related. Has anyone else encountered the same problem?

I'm am using wix 4 rc2.

**Comment by robmen** [comment]
A [verbose log file would tell all](https://support.firegiant.com/hc/en-us/articles/230912407-Create-a-log-file-). But if you disable a feature during uninstall, this is the sort of behavior you'd see.

**Comment by erikhermansson79** [comment]
I attached a log file for each scenario:
* msiexec /x  [uninstall_x.txt](https://github.com/wixtoolset/issues/files/10517587/uninstall_x.txt)
* msiexec /i -> "Remove" [uninstall_i_remove.txt](https://github.com/wixtoolset/issues/files/10517585/uninstall_i_remove.txt)
* msiexec /i -> "Change" -> disable feature. [uninstall_i_change_disable.txt](https://github.com/wixtoolset/issues/files/10517656/uninstall_i_change_disable.txt)

**Comment by fyodorkor** [comment]
HI,
It seems that you have duplicated the following properties and folders:
INSTALLFOLDER
WEBSERVER
WWWROOT
NEWWEBSITEFOLDER
 SV
EN
 EN_LOCALES
 LOCALES
 SV_LOCALES
 CSS
 STATIC
 JS
MEDIA
TARGETDIR

All directories identifiers can be referenced as properties and therefore
there is no need to add them to the properties table. Remove above from
properties table, rebuild and try again

On Fri, Jan 27, 2023 at 11:57 AM Erik Hermansson ***@***.***>
wrote:

> I attached a log files for each scenario:
>
>    - msiexec /x
>    - msiexec /i -> "Remove"
>    - msiexec /i -> "Change" -> disable feature.
>
> uninstall_x.txt
> <https://github.com/wixtoolset/issues/files/10517587/uninstall_x.txt>
> uninstall_i_remove.txt
> <https://github.com/wixtoolset/issues/files/10517585/uninstall_i_remove.txt>
> uninstall_i_change_disable.txt
> <https://github.com/wixtoolset/issues/files/10517656/uninstall_i_change_disable.txt>
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/orgs/wixtoolset/discussions/7180#discussioncomment-4795323>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ACCKQ27CW62PN3UNHSKDT7TWUOLYPANCNFSM6AAAAAAUHJXCCY>
> .
> You are receiving this because you are subscribed to this thread.Message
> ID: ***@***.***>
>


-- 
Regards,
Fyodor Koryazhkin..

**Reply by erikhermansson79** [reply]
I haven't added them as properties, only as folders. But even if I had, why would the behaviours differ when running /i -> "Remove" vs /x?

**Comment by fyodorkor** [comment]
If you run /x you do not enter UI sequence and those folders are populated
properly.
Open your package with Orca and check Properties table

On Mon, Jan 30, 2023 at 11:14 AM Erik Hermansson ***@***.***>
wrote:

> I haven't added them as properties, only as folders. But even if I had,
> why would the behaviours differ when running /i -> "Remove" vs /x?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/orgs/wixtoolset/discussions/7180#discussioncomment-4816442>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ACCKQ25765K7ERY6U4PFWTTWU6BA5ANCNFSM6AAAAAAUHJXCCY>
> .
> You are receiving this because you commented.Message ID:
> ***@***.***>
>


-- 
Regards,
Fyodor Koryazhkin..

**Comment by erikhermansson79** [comment]
My properties table looks like this:
<img width="551" alt="Skärmbild 2023-01-30 124704" src="https://user-images.githubusercontent.com/26566646/215713839-d2299e45-6beb-498d-9d4b-645776f0f97c.png">

But your comments led me in another direction. I had previously switched from manually adding some files to using directory harvesting, but I had kept the old directory definitions. When I removed them I noticed that I was trying to use a property as a directory. After correctly setting the directory I saw new errors in my log:

`Disallowing uninstallation of component: {GUID} since another client exists`

I usually test my MSI package in Windows sandbox, but there is no way to install IIS there so I tested on my main system. There must have been remains from an old version left, and after purging the components from the registry everything seems to work now.

Thank you very much for your help!
